File: study_code/src/programmers/codeTest/javasript/codeTest_MaxNum.js
/**
 * 
 */


function solution(numbers) {    
     var answer = numbers.map(v=>v+'')  // 각숫자들 문자로변환
                        .sort((a,b) => (b+a) - (a+b))  // 문자로 변환한 숫자 연결하여 비교정렬
                        .join(''); // 문자열로  합치기 
    return answer[0]==='0'?'0':answer; //배열이 0으로만 구성될 경우 0만 출력 
}

File: study_code/src/programmers/codeTest/java/codeTest_MaxNum.java
package programmers.codeTest.java;

import java.util.Arrays;
import java.util.Comparator;
import java.util.stream.Stream;

public class codeTest_MaxNum {
    public static String solution(int[] numbers) {
        String [] str = Arrays.toString(numbers).substring(1,Arrays.toString(numbers).length()-1 ).split(", ");
        String answer ="";
        
        Arrays.sort(str, new Comparator<String>() {
        	public int compare(String o1, String o2) {
        		return (o2+o1).compareTo(o1+o2);
        		}
		});

        
        if(str[0].startsWith("0")) {
        	answer = "0";
        }else {
        	answer= Arrays.toString(str).substring(1,Arrays.toString(numbers).length()-1 ).replace(", ", "");
        }
        
        return answer;
    }
    
    public static void main(String[] args) {
    	int numbers[] = {3, 30, 34, 5, 9};
    	solution(numbers);
    	
	}
}
